swift-tools-support-core/Sources/TSCTestSupport/PseudoTerminal.swift

```diff
@@ -31,10 +31,17 @@ public final class PseudoTerminal {
     }
     
     public func readMaster(maxChars n: Int = 1000) -> String? {
-        var buf: [CChar] = [CChar](repeating: 0, count: n)
-        if read(master, &buf, n) <= 0 {
-            return nil
+        let buf: [CChar] = .init(unsafeUninitializedCapacity: n + 1 /* +1 for null terminator */ ) {
+            buf, initializedCapacity in
+            initializedCapacity = read(master, &buf, n)
```

Comment:
This should actually be `initializedCapacity = read(master, buf.baseAddress, n)`, because otherwise you're actually overwriting the pointer itself.  The reason the code doesn't just copy over is that in the old code `buf` was an array, but in the new code `buf` is a `UnsafeMutableBufferPointer`.